nhs/data/geography.py

```diff
@@ -0,0 +1,295 @@
+from typing import List, Tuple  # Importing required types
+
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import polars as pl
+from loguru import logger
+from pathos.multiprocessing import ProcessingPool as Pool
+from pyproj import CRS
+
+
+def read_shapefile(path: str) -> int:
+    """Read shapefile and return total number of rows."""
+    gdf = gpd.read_file(path)
+    total_rows = len(gdf)
+    logger.info(f"Total number of rows in the shapefile: {total_rows}")
+    return total_rows
+
+
+def prepare_points(df: pl.DataFrame, crs: str) -> gpd.GeoDataFrame:
+    """
+    Prepare points data for geospatial analysis.
+
+    Parameters
+    ----------
+    df : pl.DataFrame
+        A Polars DataFrame containing at least two columns: 'LONGITUDE' and 'LATITUDE',
+        which represent geographic coordinates of the points in the data. The DataFrame
+        will be converted to a GeoDataFrame for analysis.
+    crs : str
+        The coordinate reference system (CRS) in which the output geospatial data should
+        be projected. The CRS is provided as an EPSG code (we use EPSG: 7844) as in line
+        with ABS standard. This defines how the spatial data will be interpreted in terms
+        of location, scale, and projection.
+
+    Returns
+    -------
+    gdf : gpd.GeoDataFrame
+        A GeoDataFrame with points data transformed into the specified coordinate
+        reference system (CRS). The output data will have a geometry column with Point
+        objects and be projected to the specified CRS.
+
+    Notes
+    -----
+    - CRS (Coordinate Reference System) is needed to define how spatial data (like longitude
+      and latitude) relates to the Earth's surface. We use CRS 7844 as that's what's
+      specified in the ABS 2021 datapack. As it's subject to change we'll need it in the
+      configurations.yml file for future user specification.
+    - 'EPSG:7844' is an example of a CRS in the EPSG system (European Petroleum Survey Group).
+      This is a standardized system for defining geospatial coordinates.
+    - WGS84 (EPSG:4326) is a widely used geographic CRS, representing coordinates as latitude
+      and longitude on a spherical model of the Earth.
+    """  # noqa: E501
```

Comment:
`# noqa: E501` I'm not a fan of the 79 char limit either, wondering if it can be turned off since it might conflict with `black` from time to time